proposals/xxxx-package-manager-branch-support.md

```diff
@@ -0,0 +1,50 @@
+# Package Manager Support for branches
+
+* Proposal: [SE-XXXX](xxxx-package-manager-branch-support.md)
+* Author: [Boris Bügling](https://github.com/neonichu)
+* Review Manager: TBD
+* Status: Discussion
+
+## Introduction
+
+This proposal adds enhancements to the package manifest to support development of packages without strict versioning. This is one of two features, along with "Package Manager Support for Top of Tree development", being proposed to enable use of SwiftPM to develop on "top of tree" of related packages.
+
+## Motivation
+
+The package manager currently supports packages dependencies which are strictly versioned according to semantic versioning. This is how a package's dependencies should be specified when that package is released, but this requirement hinders some development workflows:
+
+- bootstrapping a new package which does not yet have a version at all
+- developing related packages in tandem in between releases, when one package may depend on the latest revision of another, which has not yet been tagged for release
+
+## Proposed solution
+
+As a solution to this problem, we propose to extend the package manifest to allow specifying a branch instead of a version to support revlocked packages and initial bootstrapping.
+
+## Detailed Design
+
+We will introduce a second initializer for `.Package` which takes a branch instead of a version range:
+
+```swift
+import PackageDescription
+
+let package = Package(
+    name: "foo",
+    dependencies: [
+        .Package(url: "http://url/to/bar", branch: "development"),
+    ]
+)
+```
+
+Whenever dependencies are checked out or updated, if a dependency on a package specifies a branch instead of a version, the latest commit on that branch will be checked out for that package. If a dependency on a package specifies a branch instead of a version range, it will override any versioned dependencies present in the current package graph that other packages might specify.
```

Comment:
Specifically, what we would like to address here is situations where a package has changed their development dependencies in such a way that, for example, the graph is unresolvable, but they don't get an error because something is in branch mode.